File: src/main/java/gruvexp/bbminigames/menu/menus/GameMenu.java
package gruvexp.bbminigames.menu.menus;

import gruvexp.bbminigames.menu.Menu;
import gruvexp.bbminigames.twtClassic.BotBows;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.event.ClickEvent;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.Style;
import net.kyori.adventure.text.format.TextDecoration;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

public class GameMenu extends Menu {
    @Override
    public Component getMenuName() {
        return Component.text("Game Menu");
    }

    @Override
    public int getSlots() {
        return 9;
    }

    @Override
    public void handleMenu(InventoryClickEvent e) {
        Player p = (Player) e.getWhoClicked();

        if (e.getCurrentItem().getType() == Material.BOW) {
            BotBows.lobbyMenu.open(p);
        } else if (e.getCurrentItem().getType() == Material.STICK) {
            p.sendMessage(Component.text("This will be added soon. You can take a look at the arena ")
                    .append(Component.text("here", Style.style(NamedTextColor.AQUA, TextDecoration.UNDERLINED))
                            .clickEvent(ClickEvent.clickEvent(ClickEvent.Action.RUN_COMMAND, "/tp @s 38.0 31.3 -176.5 102 22"))));
        }
    }

    @Override
    public void setMenuItems() {
        ItemStack botbows = makeItem(Material.BOW, Component.text("BotBows Classic"), Component.text("The classic game of BotBows"));
        ItemStack sumo = makeItem(Material.STICK, Component.text("Sumo"), Component.text("The Sumo minigame"));
        inventory.setItem(3, botbows);
        inventory.setItem(5, sumo);
        setFillerVoid();
    }
}

File: src/main/java/gruvexp/bbminigames/menu/menus/GameMenu.kt
package gruvexp.bbminigames.menu.menus

import gruvexp.bbminigames.menu.Menu
import gruvexp.bbminigames.sumo.SumoManager
import gruvexp.bbminigames.twtClassic.BotBows
import net.kyori.adventure.text.Component
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.inventory.ItemStack

class GameMenu : Menu() {
    init {
        inventory.setItem(3, BOTBOWS)
        inventory.setItem(5, SUMO)
    }

    override fun getMenuName(): Component = Component.text("Game Menu")
    override fun getSlots(): Int = 9

    override fun handleMenu(e: InventoryClickEvent) {
        val p = e.whoClicked as Player
        val clickedItem = e.currentItem ?: return

        val action = MenuAction.valueOf(getActionId(clickedItem) ?: return)
        when (action) {
            MenuAction.JOIN_BOTBOWS -> BotBows.lobbyMenu.open(p)
            MenuAction.JOIN_SUMO -> SumoManager.sumoMenu.open(p)
        }
    }

    companion object {
        val BOTBOWS: ItemStack = makeItem(
            Material.BOW,
            Component.text("BotBows Classic"),
            MenuAction.JOIN_BOTBOWS.name,
            Component.text("The classic game of BotBows")
        )
        val SUMO: ItemStack = makeItem(
            Material.STICK,
            Component.text("Sumo"),
            MenuAction.JOIN_SUMO.name,
            Component.text("Knockback pvp with sticks")
        )
    }

    private enum class MenuAction {
        JOIN_BOTBOWS,
        JOIN_SUMO,
    }
}
